packages/rum/src/domain/segmentCollection/startDeflateWorker.ts

```diff
@@ -0,0 +1,113 @@
+import { addErrorToMonitoringBatch, display, includes, monitor } from '@datadog/browser-core'
+import { createDeflateWorker, DeflateWorker } from './deflateWorker'
+
+/**
+ * For some browser like Firefox, "new Worker()" can fail and fire an 'error' event instead of an exception,
+ * making the creation asynchronous. To handle it, we need a round trip of initialization messages to be sure
+ * the creation is correct. These worker lifecycle states handle this case.
```

Comment:
```suggestion
In order to be sure that the worker is correctly working, we need a round trip of initialization messages, making the creation asynchronous.
These worker lifecycle states handle this case.
```
it seemed a bit redundant with the other comment